wire-ios-data-model/Source/MLS/Migration/ProteusToMLSMigrationCoordinator.swift

```diff
@@ -117,6 +127,32 @@ public class ProteusToMLSMigrationCoordinator: ProteusToMLSMigrationCoordinating
         }
     }
 
+    private func finalizeMigrationIfNeeded() async throws {
```

Comment:
This method execute:
- refetching users from the BE;
- something with mix conversations.

I would split it into 2 methods. It's not very clear what `finalizeMigration` is supposed to do.

Reply:
@David-Henner What do you think? I'm confused about "...IfNeeded" in the name. We don't have any checks inside the method. Currently, when the migrationStatus is `.started` we:
 `migrateOrJoinGroupConversations()`
 `finalizeMigrationIfNeeded()`
 
 As far as I understand, the mixed conversations migration is the same and duplicated in these two methods. 
 
 Why not something like
 `migrateOrJoinGroupConversations()`
 `syncUsers()`
 ? These 2 methods can be placed inside `finalizeMigration()`.
 These are just my thoughts.

Reply:
@KaterinaWire could you have another look ? I updated the logic